app/src/main/java/ayds/lisboa/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,217 @@
+private const val FLAG_INSENSITIVE_UPPER_LOWER_CASE = "(?i)"
+private const val HTML_HTML_OPEN = "<html>"
+private const val HTML_HTML_CLOSE = "</html>"
+private const val HTML_FONT_FACE_ARIAL_OPEN = "<font face=\"arial\">"
+private const val HTML_FONT_CLOSE = "</font>"
+private const val HTML_DIV_W400_OPEN = "<div width=400>"
+private const val HTML_DIV_CLOSE = "</div>"
+private const val HTML_B_OPEN = "<b>"
+private const val HTML_B_CLOSE = "</b>"
+private const val HTML_BR = "<br>"
+private const val HTML_SPACE = " "
+
+class OtherInfoWindow : AppCompatActivity() {
+    private lateinit var artistInfoTextView: TextView
+    private lateinit var dataBase: DataBase
+    private lateinit var artistName: String
+    private lateinit var retrofit: Retrofit
+    private lateinit var lastFMAPI: LastFMAPI
+    private lateinit var openUrlButtonView: View
+    private lateinit var lastFmImageView: ImageView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView()
+        initViews()
+        initDatabase()
+        createRetrofit()
+        createLastFMAPI()
+        updateArtistInfoView()
+    }
+
+    private fun initViews() {
+        lastFmImageView = findViewById(R.id.imageView)
+        artistInfoTextView = findViewById(R.id.artistInfoTextView)
+        openUrlButtonView = findViewById(R.id.openUrlButton)
+    }
+
+    private fun setContentView() {
+        setContentView(R.layout.activity_other_info)
+    }
+
+    private fun initDatabase() {
+        dataBase = DataBase(this)
+    }
+
+    private fun updateArtistInfoView() {
+        updateArtistName()
+        searchArtistInfo()
+    }
+
+    private fun updateArtistName() {
+        artistName = intent.getStringExtra(ARTIST_NAME_EXTRA).toString()
+    }
+
+    private fun createRetrofit() {
+        retrofit = Retrofit.Builder()
+            .baseUrl(URL_BASE_API)
+            .addConverterFactory(ScalarsConverterFactory.create())
+            .build()
+    }
+
+    private fun createLastFMAPI() {
+        lastFMAPI = retrofit.create(LastFMAPI::class.java)
+    }
+
+    private fun searchArtistInfo() {
+        Thread {
+            updateArtistInfo()
+        }.start()
+    }
+
+    private fun updateArtistInfo() {
+        val biography = getArtistBiography()
+        when(biography.isLocallyStored) {
+            false -> setUrlButton(biography.url)
+            else -> {}
+        }
+        updateViewInfo(biography)
+    }
+
+    private fun getArtistBiography(): Biography {
+        val artistInfo = getArtistInfoFromDataBase()
+        var artistBiography = Biography(artistInfo, DEFAULT_STRING, true)
+        if(artistBiography.artistInfo.isEmpty()){
+            artistBiography = getArtistBiographyFromLastFMAPI()
+            if(artistBiography.artistInfo.isNotEmpty())
+                saveArtistInfoInDataBase(artistBiography.artistInfo)
+        }
+        return artistBiography
```

Comment:
No tendríamos que crear el objeto Biography si no estaba en la db. Mejor hacer:
```suggestion
        val artistInfo = getArtistInfoFromDataBase()
        val artistBiography = 
                if(artistBiography.artistInfo.isEmpty()){
                   getArtistBiographyFromLastFMAPI().apply {
                         if(artistBiography.artistInfo.isNotEmpty())
                              saveArtistInfoInDataBase(artistBiography.artistInfo)
                   }
                } else {
                    Biography(artistInfo, DEFAULT_STRING, true)
                } 
        return artistBiography
```